frontend/libs/studio-components/src/components/StudioModal/StudioModalDialog/StudioModalDialog.module.css

```diff
@@ -0,0 +1,39 @@
+.dialog {
+  overflow: hidden; /* Avoid double scrollbars; overflow is handled by Modal.Content */
+  display: flex;
+  flex-direction: column;
+}
+
+.heading {
+  border-bottom: 1px solid var(--fds-semantic-border-divider-default);
+  padding: var(--studio-modal-spacing);
+}
+
+.heading > * {
+  align-items: center;
+  display: flex;
+  line-height: 1;
+}
+
+.icon {
+  display: flex;
+  margin-right: var(--studio-modal-spacing);
+}
+
+.icon > * {
+  transform: scale(1.5);
+}
+
+.content {
+  padding: 0;
+  flex: 1;
```

Comment:
I'm not sure if it's the best solution, but deleting this line seems to fix the height issue
```suggestion
```

Reply:
Nice catch! The reason for the `flex: 1` there is to make the `.content` element take up the full height. Without it, the settings modal looks like this:
![image](https://github.com/user-attachments/assets/48ee98f9-0a87-4ad4-ba45-4d5176e1242d)
Notice that the grey area does not reach down to the bottom.

However, since we know that this causes problems for Safari, I'll see if I can find another solution without using `flex`.

Reply:
@TomasEng I've tried different solutions, but the one that worked best on both Safari and Chrome was to use `flex: 1` on the `.content` class, as you did, but only for the settings modal, as it appears to work well on Safari only if the modal has a height set (e.g., height: 80vh).
A solution that might not be the most optimal but that works could be to apply `flex: 1` to the `.content` class only for the settings modal by passing something like a `contentClassName`:

https://github.com/Altinn/altinn-studio/compare/ny-modal...ny-modal-test

Reply:
Another solution could be to pass the header and the content as children to the modal, but this would require more work 🤷‍♂️ :

```
<StudioModal>
        <StudioModal.Header icon={<TrashIcon />} closeButtonTitle={}>...</StudioModal.Header>
        <StudioModal.Content className={classes.modalContent}>...</StudioModal.Content>
</StudioModal>
```

Reply:
Although the second suggestion is more aligned with how it is done in The Design System, I think it's too verbose for our use cases. It will just make the code harder to read. Therefore I think the `contentClassName` prop is a much better option.

How will the scrolling appear in the modals without `flex: 1`? If I remember correctly, I had some issues with the scrolling when I tried to remove it (which is why I reverted my first fix).

Reply:
Agreed. The scrolling seems to work as exepected. Here is how it looks on both Safari and Chrome:

## Safari

### Modals without `flex: 1`

<table>

<tr>
<td>

https://github.com/user-attachments/assets/d13176d3-4526-4c55-8c41-14a37d4f6e3d

</td>
<td>

https://github.com/user-attachments/assets/5f2d32c5-2ef6-407e-8da9-20e4b6b99a55

</td>
<td>

https://github.com/user-attachments/assets/cff5a376-6ede-49ed-8766-78da64ae8ba9

</td>
</tr>
</table>

### Modal with `flex: 1`

The sidebar on the left gets an extra horizontal scrollbar, but this issue doesn't come from this PR, as the scrollbar is also visible with the old modal in production. I tried to fix it, but I couldn't find a solution that works well on both Chrome and Safari 🤷‍♂️ 

https://github.com/user-attachments/assets/bab19bea-7c66-4d94-8831-d9b43d27d9e7

## Chrome

### Modals without `flex: 1`

<table>

<tr>
<td>

[chrome-clone.webm](https://github.com/user-attachments/assets/d65bf49f-a4bc-40e4-ada6-d3dfd64bc3d3)

</td>
<td>

[chrome-diff.webm](https://github.com/user-attachments/assets/4eb498cc-4831-4858-a167-41db161efbe7)

</td>
<td>

[chrome-changes.webm](https://github.com/user-attachments/assets/1e6e9583-9514-42fc-a44e-1226f873273b)

</td>
</tr>
</table>

### Modal with `flex: 1`

[chrome-settings.webm](https://github.com/user-attachments/assets/94f81788-23d5-40a5-a6f4-3786381d2e97)

Reply:
Great work! I'll merge it in.